locals.tf

```
@@ -43,6 +43,11 @@ locals {
       enable             = true
       serviceAccountName = local.secrets_store_csi_driver_service_account
     } : null
+    csiSecretsStoreProviderAws = var.enable_csi_secrets_store_provider_aws && var.enable_csi_secrets_store_provider_aws_gitops ? {
+      enable             = true
+      serviceAccountName = local.csi_secrets_store_provider_aws_service_account
+    } : null
```

The serviceAccountName is not needed to be passed. The service account needs to be IRSA annotated for the workload consuming the secret not the csi driver here.

```suggestion
    csiSecretsStoreProviderAws = var.enable_csi_secrets_store_provider_aws && var.enable_csi_secrets_store_provider_aws_gitops ? {
      enable = true
    } : null
```